### PR TITLE
New api and features

### PR DESCRIPTION
**Pull request by salomonl**
This branch follows the C api simplification of Nomad:
- update C wrappers functions and fix version compatibility
- add new (and tested) Nomad options

### CODE REVIEW

File: src/c_wrappers.jl [hunk 1]
```diff
@@ -51,95 +51,81 @@ function create_c_nomad_problem(eval_bb::Function,
...
 
-function set_nomad_nm_search!(prob::C_NomadProblem, nm_search::Bool)
-    is_ok = ccall((:setNomadUseCache, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, nm_search)
+function add_nomad_array_of_double_param!(prob::C_NomadProblem, keyword::String, a::Vector{Float64})
+    is_ok = ccall((:addNomadArrayOfDoubleParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}, Ptr{Float64}), prob.ref, keyword, a)
     return is_ok
 end
 
-function solve_problem(prob::C_NomadProblem, x0::Vector{Float64})
+function solve_nomad_problem(prob::C_NomadProblem, x0s::Vector{Float64}, nb_starting_points::Int)
```
Comment (amontoison) [inline src/c_wrappers.jl:128]: Pourquoi est-ce que tu as changé `x0` en `x0s` ?

Comment (salomonl) [reply src/c_wrappers.jl:128]: C style: you can now pass to the C function several starting points, hence the `x0s`. I have not put this feature yet on the core functions.

**Comment by amontoison** [comment]
I rebased your branch. I propose to merge this PR if all tests passed.

**Comment by amontoison** [comment]
@salomonl, three new features are not tested : https://github.com/bbopt/NOMAD.jl/pull/32/checks?check_run_id=1920004063
We can improve the code coverage with another PR.

**Pull request merged**